client/my-sites/design-preview/index.js

```diff
@@ -157,12 +158,14 @@ const DesignPreview = React.createClass( {
 } );
 
 function mapStateToProps( state ) {
-	const selectedSiteId = getSelectedSiteId( state );
+	const selectedSite = getSelectedSite( state );
+	const selectedSiteId = selectedSite ? selectedSite.ID : null;
 	if ( ! state.preview || ! state.preview[ selectedSiteId ] ) {
 		return { selectedSiteId };
 	}
 	const { previewMarkup, customizations, isUnsaved } = state.preview[ selectedSiteId ];
 	return {
+		selectedSite,
```

Comment:
You'll probably also want to return `selectedSite` in the early return a few lines up.